### ISSUE TITLE
Reversed Viewport broken on nVidia and AMD

### ISSUE THREAD

**Issue by pizuz** [KhronosGroup/MoltenVK#955]
Hello,

since #230 MoltenVK is supposed to support reversed viewport depth. However, enabling this in Dolphin causes glitches on both nVidia and AMD adapters and has therefore been [disabled since on macOS](https://github.com/dolphin-emu/dolphin/pull/8259). Intel Iris seems to be unaffected. A combination of reversed depth range and depth clamping was suspected, at first, but this glitch is triggered independently.

Affected cards to my knowledge:
[nVidia GeForce GT 750M](https://github.com/dolphin-emu/dolphin/pull/8259#issuecomment-650589132)
[AMD Radeon Pro 455](https://github.com/dolphin-emu/dolphin/pull/8244#issuecomment-513514100)

Unaffected card:
[Intel Iris Graphics 550](https://github.com/dolphin-emu/dolphin/pull/8259#issuecomment-610384050)

Example screenshot below. The ship is supposed to render in front of the clouds, the sun behind it. This happens on both the latest versions of MoltenVK (1.0.43) and gfx-portability (0.7).

![Screenshot 2020-06-27 at 19 25 07](https://user-images.githubusercontent.com/12557040/85928288-3ae20c80-b8ac-11ea-8144-2ab28fdc14a7.png)

You can either build dolphin yourself with this the changes below, or use [this build](https://dl.dolphin-emu.org/builds/f9/5b/dolphin-master-5.0-10709.dmg).

```diff
diff --git a/Source/Core/VideoCommon/DriverDetails.cpp b/Source/Core/VideoCommon/DriverDetails.cpp
index 1670f1a028..ac98143311 100644
--- a/Source/Core/VideoCommon/DriverDetails.cpp
+++ b/Source/Core/VideoCommon/DriverDetails.cpp
@@ -112,8 +112,8 @@ constexpr BugInfo m_known_bugs[] = {
      -1.0, -1.0, true},
     {API_VULKAN, OS_ALL, VENDOR_QUALCOMM, DRIVER_QUALCOMM, Family::UNKNOWN,
      BUG_BROKEN_REVERSED_DEPTH_RANGE, -1.0, -1.0, true},
-    {API_VULKAN, OS_OSX, VENDOR_ALL, DRIVER_PORTABILITY, Family::UNKNOWN,
-     BUG_BROKEN_REVERSED_DEPTH_RANGE, -1.0, -1.0, true},
+    //{API_VULKAN, OS_OSX, VENDOR_ALL, DRIVER_PORTABILITY, Family::UNKNOWN,
+    // BUG_BROKEN_REVERSED_DEPTH_RANGE, -1.0, -1.0, true},
     {API_VULKAN, OS_ALL, VENDOR_ARM, DRIVER_ARM, Family::UNKNOWN, BUG_SLOW_CACHED_READBACK_MEMORY,
      -1.0, -1.0, true},
     {API_VULKAN, OS_ALL, VENDOR_QUALCOMM, DRIVER_QUALCOMM, Family::UNKNOWN,
```

**Comment by cdavis5e** [comment]
> since #230 MoltenVK is supposed to support reversed viewport depth.

I think you might be a bit confused. `VK_KHR_maintenance1` only enables negative viewport height. It subsumed the `VK_AMD_negative_viewport_height` extension, which also did that, and which was already implemented prior to #230 anyway. `minDepth >= maxDepth` is part of the base 1.0 spec.

Anyway, I'm not sure what we can do. This is probably a Metal driver bug, since MoltenVK just passes the viewports through to Metal.

**Comment by pizuz** [comment]
Makes sense, thanks. From what I gather, there were driver bugs like that on Windows, as well, but those have since been fixed by nVidia and AMD. Closing.
